data_structures/test_stack.py

```diff
@@ -0,0 +1,23 @@
+from stack import Stack as ST
+import pytest
+
+
+def test_load():
+    teststack = ST([1, 2])
+    assert teststack.top.val is 1
+
+
+def test_empty_stack_has_no_top(empty_stack):
+    assert empty_stack.top is None
+
+
+def test_insertion(empty_stack):
+    assert empty_stack.top is None
+    assert empty_stack.push(1).val == 1
+
+
+def test_empty_val_on_insert(empty_stack):
+    with pytest.raises(TypeError):
+        empty_stack.push()
```

Comment:
This is a unnecessary test as it is only testing if Python is working properly. The Python interpreter itself will raise a `TypeError` if a function is called with too many or too few required arguments without ever calling the function.